src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax;
+
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+
+public class Main {
+    private static final Logger logger = LogManager.getLogger(Main.class);
+
+    public static void main(String[] args) {
+        if (args.length < 2) {
+            logger.error("Please provide both source and target file paths as arguments.");
+            return;
+        }
+
+        final String sourceFilePath = args[0];
+        final String targetFilePath = args[1];
+
+        TransactionProcessingService transactionProcessingService =
+                new TransactionProcessingService();
+        try {
+            transactionProcessingService.processTransactions(sourceFilePath, targetFilePath);
+            logger.info(String.format("Report was successfully written to: %s", targetFilePath));
+        } catch (Exception e) {
```

Comment:
Remember to follow the checklist item regarding exception handling. It's advised to avoid leaving generic catch blocks. Instead, catch specific exceptions and handle them accordingly. If a general exception is to be caught, it should be re-thrown as a RuntimeException with an informative message.